### ISSUE TITLE
Add App Block Support to additional sections

### ISSUE THREAD

**Issue by nicklepine** [Shopify/dawn#38]
Let's support app blocks in more sections. This ticket will encompass adding support for app blocks to sections that meet the following requirements:
- The section doesn't require a design rework in order to be extended to support app blocks
- We foresee these sections being desired insertion points for apps. 

Sections to tackle: 

- [x]  Cart template - Subtotal section
- [x]  Blog Post template
- [x]  Password template - Newsletter section

Can't be done:

Announcement Bar

TL;DR - it's a static section and the app blocks rely on JSON templates to work.  App blocks, therefore, can't be added in the announcement bar, header, or footer.
